### PR TITLE
DAOS-6726 mgmt: Fix mgmt_svc stop

### PR DESCRIPTION
**Pull request by liw**
ds_mgmt_svc_stop has a few problems that this patch addresses:

  - It is operating on the primary xstream, rather than xstream 0. This
    is dangerous because rsvc is not supposed to be used outside of
    xstream 0 (e.g., reference counters are not pthread safe).

  - It leaks the mgmt_svc reference held by the rsvc hash table.
    Although rsvc's module fini callback releases the reference, it
    happens late, after the xstreams are terminated. mgmt should clean
    up the mgmt_svc object before its module cleanup callback completes.

  - It does not wait for map_distd to stop. This causes the last rsvc
    referenece to be freed late and unnecessarily racy.

Signed-off-by: Li Wei <wei.g.li@intel.com>

### CODE REVIEW

**Comment by liw** [comment]
Previous failures look like due to the CI env maintenance.

File: src/mgmt/srv_system.c [hunk 1]
```diff
@@ -343,37 +343,31 @@ ds_mgmt_svc_start(void)
...
 }
 
+static void
+stopper(void *arg)
+{
+	int rc;
+
+	rc = ds_rsvc_stop_nodb(DS_RSVC_CLASS_MGMT, &mgmt_svc_id);
+	if (rc != 0)
+		D_DEBUG(DB_MGMT, "ignoring "DF_RC"\n", DP_RC(rc));
```
Comment (ashleypittman) [inline src/mgmt/srv_system.c:353]: I wonder if this should be D_ERROR().

**Review by ashleypittman — APPROVED** [review]

File: src/mgmt/srv_system.c (hunk 1 above)
Comment (liw) [reply src/mgmt/srv_system.c:353]: In an engine who hasn't started mgmt_svc, rc will be -DER_ALREADY when we reach here. I assume NLT would complain if we logged this as an error, no?

**Review by dmiter — APPROVED** [review]

File: src/mgmt/srv_system.c (hunk 1 above)
Comment (ashleypittman) [reply src/mgmt/srv_system.c:353]: As long as NLT is consistent then there's no problem there.

If there are legitimate reasons this code could be reached then it's not an error so D_DEBUG() is correct.

**Pull request merged**